cpp/test/adapter-non-local.test.js

```diff
@@ -1,268 +1,671 @@
+    contract: null,
+  }
+
+  const fromEvmXerc20 = {
+    symbol: `X${evmTokenSymbol}`,
+    precision: fromEvmXerc20Precision, // different from the underlying token - only uint64 is supported by Asset type
+    account: `x${evmTokenSymbol.toLowerCase()}.token`,
+    maxSupply: `${trimPrecision(maxSupply, fromEvmXerc20Precision)} X${evmTokenSymbol}`,
+    minFee: `${trimPrecision('0.0018', fromEvmXerc20Precision)} X${evmTokenSymbol}`,
+    contract: null,
+  }
+
+  const wrongToken = {
+    symbol: 'WRG',
+    account: 'wrg.token',
+    maxSupply: `${maxSupply} WRG`,
+    bytes: no0x(
+      ethers.zeroPadValue(
+        ethers.toBeHex(getSymbolCodeRaw('0.0000 WRG').toString()),
+        32,
+      ),
+    ),
+    contract: null,
+  }
+
+  const lockbox = {
+    account: 'lockbox',
+    contract: null,
+  }
+
+  const adapter = {
+    account: 'adapter',
+    contract: null,
+  }
+
+  const notInitAdapter = {
+    account: 'adapter2',
+    contract: null,
+  }
+
+  const receiver = {
+    account: 'receiver',
+    contract: null,
+  }
+
+  const blockchain = new Blockchain()
+
+  const user = 'user'
+  const evil = 'evil'
+  const issuer = 'issuer'
+  const bridge = 'bridge'
+  const recipient = 'eosrecipient'
+  const feemanager = 'feemanager'
+
+  before(async () => {
+    blockchain.createAccounts(user, evil, issuer, bridge, recipient, feemanager)
+    lockbox.contract = deploy(
+      blockchain,
+      lockbox.account,
+      'contracts/build/lockbox',
+    )
+    fromEvmXerc20.contract = deploy(
+      blockchain,
+      fromEvmXerc20.account,
+      'contracts/build/xerc20.token',
+    )
+    adapter.contract = deploy(
+      blockchain,
+      adapter.account,
+      'contracts/build/adapter',
+    )
+    notInitAdapter.contract = deploy(
+      blockchain,
+      notInitAdapter.account,
+      'contracts/build/adapter',
+    )
+    receiver.contract = deploy(
+      blockchain,
+      receiver.account,
+      'contracts/build/test.receiver',
+    )
+  })
+
+  const setup = async () => {
+    await fromEvmXerc20.contract.actions
+      .create([issuer, fromEvmXerc20.maxSupply])
+      .send(active(fromEvmXerc20.account))
+
+    const mintingLimit = `${trimPrecision('1000', fromEvmXerc20Precision)} ${fromEvmXerc20.symbol}`
+    const burningLimit = `${trimPrecision('600', fromEvmXerc20Precision)} ${fromEvmXerc20.symbol}`
+
+    await fromEvmXerc20.contract.actions
+      .setlimits([adapter.account, mintingLimit, burningLimit])
+      .send(active(fromEvmXerc20.account))
+  }
+
+  describe('adapter::create', () => {
+    it('Should throw if called by not authorized account', async () => {
+      const action = adapter.contract.actions
+        .create([
+          fromEvmXerc20.account,
+          precision(fromEvmXerc20.precision, fromEvmXerc20.symbol),
+          evmUnderlyingToken.account,
+          precision(evmUnderlyingToken.precision, evmUnderlyingToken.symbol),
```

Comment:
We don't need precision on for the token symbol on non-local chains, I'd stress that with
```suggestion
          precision(0, evmUnderlyingToken.symbol),
```

Reply:
...and on create's relates tests

Reply:
a symbol always carries the precison too or it is not valid. 
Being the symbol defined as
`symbol = TST`, passing it just like that would throw. It would be valid if it was defined as `symbol = P, TST` where P is the precision (in this case it would be `TST = 8, TST`), but for semplicity I use the `precision` function in order to avoid the need of another function that generate the correct string starting from precision and symbol name.

Reply:
so passing `0,TST` will throw? I didn't get your point, sorry

Reply:
I mean, apply exactly my suggestion above and you see that the test unit will pass

Reply:
passing `0,TST` throws on the suggested test. Am I missing something?